src/main/kotlin/anttracker/MainMenuScreen.kt

```diff
@@ -1,24 +1,18 @@
 package anttracker
 
-val mainMenu =
-    screenWithMenu {
-        title("Main menu")
-        option("Issues") { issuesMenu }
-        option("Requests") { requestsMenu }
-        option("Products") { productsMenu }
-        option("Contacts") { contactsMenu }
-        content { t -> t.printLine("This is the main menu") }
+val requestsMenu =
+    object : Screen {
+        override fun run(): Screen? {
+            TODO("Not yet implemented")
+        }
     }
 
-private val requestsMenu =
-    screenWithMenu {
-        content { t -> t.printLine("We are in the requests menu") }
-    }
-private val productsMenu =
-    screenWithMenu {
-        content { t -> t.printLine("We are in the products menu") }
-    }
-private val contactsMenu =
-    screenWithMenu {
-        content { t -> t.printLine("We are in the contacts menu") }
+val mainMenu =
+    object : Screen {
+        val options: Map<String, Screen> = mapOf("Issue" to issuesMenu, "Requests" to requestsMenu)
```

Comment:
The Main.kt file should have the main() function implemented like so:

```kotlin
fun main() {
    while (true) {
        val mainMenuText =
            """
            == MAIN MENU ==
            1   View/Edit Issue
            2   New request
            3   New release
            4   New contact
            5   New product
            
            Please select a command. ` to exit program: 
            """.trimIndent()
        print(mainMenuText)

        when (val selection = readln()) {
            "`" -> break
            "1" -> issueMenu()
            "2" -> requestMenu()
            "3" -> releaseMenu()
            "4" -> contactMenu()
            "5" -> productMenu()
            else -> {
                println("Bad input: $selection.")
            }
        }
    }
}
```

So if you would like to use these interfaces I believe it should be kept in Issue.kt & Issues.kt, as discussed in our meeting. Otherwise you'll need to teach / enforce / make sure Tyrus, Angela, and myself understand how to use these interfaces, which I don't think is a valuable use of our time

Reply:
Also, I believe this file is redundant, it's logic is better kept in Main.kt, if necessary.

Reply:
Please, could you clarify which file is redundant?
Also, here is the implementation of your menu using a screen (that is actually just a run function).
```kotlin
val mainMenu =
    object : Screen {
        val mainMenuText =
            """
            == MAIN MENU ==
            1   View/Edit Issue
            2   New request
            3   New release
            4   New contact
            5   New product
            
            Please select a command. ` to exit program: 
            """.trimIndent()

        override fun run(): Screen? {
            print(mainMenuText)

            val selection = readln()

            return when (selection) {
                "1" -> issueMenu
                "2" -> requestMenu
                "3" -> releaseMenu
                "4" -> contactMenu
                "5" -> productMenu
                else -> mainMenu
            }
        }
    }

```

Reply:
The file I was referring to is the one I made this comment on: MainMenuScreen.kt.

Also, thanks for the example, but I want to stress again that this enforces the Screen interface on all of the submenus, something that we decided we **did not want to do**. So please keep Main.kt as discussed.

Reply:
As I have stated previously, I am not enforcing anything on anyone. This was only meant to demonstrate the usage of a `Screen` for modeling a menu.

Reply:
Eitan, I am referring to your code snippit:

```kotlin
val mainMenu =
    object : Screen {
        val mainMenuText =
            """
            == MAIN MENU ==
            1   View/Edit Issue
            2   New request
            3   New release
            4   New contact
            5   New product
            
            Please select a command. ` to exit program: 
            """.trimIndent()

        override fun run(): Screen? {
            print(mainMenuText)

            val selection = readln()

            return when (selection) {
                "1" -> issueMenu
                "2" -> requestMenu
                "3" -> releaseMenu
                "4" -> contactMenu
                "5" -> productMenu
                else -> mainMenu
            }
        }
    }
```

Which enforces that issueMenu, requestMenu, releaseMenu, contactMenu, productMenu, all return Screens. This is enforcing the Screen interface on those menus. That's what I'm referring to.

Reply:
That is what I had in mind for modelling the menus as screens.

Reply:
When you said I was enforcing the `Screen` type on all the other modules, I thought that you meant that what I was suggesting was not optional. My earlier response was based on that interpretation. This is a suggestion for how we could model the screens and is not final.